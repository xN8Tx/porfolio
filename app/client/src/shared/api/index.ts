export { publicHTTP } from "./public-http";
