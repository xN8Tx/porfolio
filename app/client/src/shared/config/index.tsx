export const availableThemes = ["light", "dark"];
