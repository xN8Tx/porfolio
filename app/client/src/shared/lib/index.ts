export * from "./cn";
export * from "./add-zero-to-num";
