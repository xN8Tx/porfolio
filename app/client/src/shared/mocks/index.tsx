import { render } from "@testing-library/react";
import { Children } from "react";

export const mockStrapiImage = {
  id: 1,
  documentId: "l73ns0nchtgvjgk49ti5vu9c",
  name: "whitecrow.webp",
  alternativeText: null,
  caption: null,
  width: 2012,
  height: 1695,
  hash: "whitecrow_26b66602d4",
  ext: ".webp",
  mime: "image/webp",
  size: 30.67,
  url: "/uploads/whitecrow_26b66602d4.webp",
  previewUrl: null,
  provider: "local",
  provider_metadata: null,
  blurHash:
    "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAEAElEQVR4AcXBQWIbxgEEwZ4l/v9fAjs9AcQodk7JyarK3viDDn/Y4Q87/EEDHvwhAgMOf4DAgAEP/mECAmNs48E/SECGihvbePAPEehERUXHGA/+AZ2oVFFRcWMbD/6n8d/C/2sTHbVUUami4sY2HmX8svHLxnjb+GUj/AiQhPCWAIHwFggw3gYbbqhUUanSlipuTHHjcV8vfhlv42Mb4W3jI/wIEEICSQhvCX+3gRtOVFSqVKnSihOVbTxerxeMt/HLeBvhbeMjQIAkBEhCeEsIbwkf2xiwDZUqKlWqtKWKihvbeLyeT/5jI/zbIIzwI4QEDpCEJHwk4bdtDFBxo0qVKm2pUkXFyQaP1/PJb+Ft4yNAgAABAiRBQgJJCG8Jv21jG26o1FFLlVpuRaWKG9t4vJ5P/i6DMD4CJOEAAZJwgCQk4SP8GDBgGyrdUGnLVWpppUoVJxs8nt/fhL8ECBAgCQEOkIQDmJCEAEn4Oze24UaVKm25SluuUouKDjcer+eT3wIECBAgCQc4CQFOQhIOkISPJIzBYBtudEOllltpy1VuS5UqTrbxeH5/8xF+JOEAAU7CAZJwEk7gEExIQvjLNrbhRidVWrlKW27LVdpSxQ03Hs/vb8JbIIQAJxDCSTiBQzgJJ+EknEAIvwQYDNiGSpUqV2nLbbmWW7mWVtzYxuP1/OYjhAAJHMJJOAkn4QS+EpYwgglJCD8GbGMbKlWqXMut3JbbcluupYoONx7P7ycfARI4hJNwEk7CV8JXwhJOwkk4CeEtEGCDbbihUuUqt+W23Jbb8mqp5So6tvF4fX/zkUAIJ+EkfCWchCXshJPDV8ISTDj8NzdUnLRyldtyW14tt5dXSyvXosONx/P5TYAQknACXzmY8HXCcljC1wnksIQkjLcA4224oaNKLbdyW27L7eV1y225lltxouPxej4JEMJJOAkmfJ1ADpwDCZwDCUs4CQtvAQYDJzqqtKWW23Jvub3cW14tt+VaVNx43OcTCAmchK+Erxw4gRw4B86BE8hhCUtIILwNtuGGSpVabktvub3cW24vt+W23JYqbjzu68XHSVjCEnYO5MA55Bw4h5yQHEggIYHwNhhDh4qWVmxpL72X3nJ7ubfclmup4sbjvi4JSDgJXydwDskh5+DXITvoIUfIgYQEwttgG05UWrGlLe2l99J76S3tpS1tqeLGw14+TgIJ8ZAzPEOGjGx4RnbIGUuAMD7GHE5UVrHFXrzFFlvsxXtpS1uquPGwBQIBcsgZYcgQEDjAgAVGWAKEH2OOTaaoqEyZxRZbbFGxxRYVNx46kiEhyDyMsYxNZhhjjJ2xwfgRBoNtsDEHEyZzbGMb29jGlE22sckc/wJBOGkmYfYa3wAAAABJRU5ErkJggg==",
};
