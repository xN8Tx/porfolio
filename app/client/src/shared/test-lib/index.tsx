import { NextIntlClientProvider } from "next-intl";
import { ThemeProvider } from "next-themes";

import { render as renderRTL } from "@testing-library/react";

import defaultMessages from "../../../messages/en.json";
import { Suspense } from "react";

const Providers = ({ children }: { children: React.ReactNode }) => (
  <Suspense fallback={<></>}>
    <NextIntlClientProvider messages={defaultMessages} locale="en">
      <ThemeProvider defaultTheme="dark">{children}</ThemeProvider>
    </NextIntlClientProvider>
  </Suspense>
);

export const render = (ui: React.ReactElement, options = {}) =>
  renderRTL(ui, { wrapper: Providers, ...options });
