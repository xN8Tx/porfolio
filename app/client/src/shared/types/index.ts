export * from "./api";
export * from "./strapi";
export * from "./next";
