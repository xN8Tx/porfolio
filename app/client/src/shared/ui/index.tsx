export * from "./paragraph";
export * from "./heading";
export * from "./button";
export * from "./image";
export * from "./link";
export * from "./svg";
