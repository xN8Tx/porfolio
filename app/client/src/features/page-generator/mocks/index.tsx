import { mockStrapiImage } from "@/shared/mocks";

export const mockStrapiHero = {
  id: 1,
  documentId: "1",
  title: "Evgeniy Shteyn",
  subtitle: "Frontend dev.",
  __component: "dynamic-zone.hero",
  locale: "en",
  componentId: "hero",
};

export const mockStrapiSkills = {
  skills: [
    { id: 1, locale: "en", documentId: "1", title: "HTML" },
    { id: 2, locale: "en", documentId: "2", title: "JavaScript" },
    { id: 3, locale: "en", documentId: "3", title: "TypeScript" },
    { id: 4, locale: "en", documentId: "4", title: "NextJS" },
    { id: 5, locale: "en", documentId: "5", title: "TailwindCSS" },
    { id: 6, locale: "en", documentId: "6", title: "Zustand" },
    { id: 7, locale: "en", documentId: "7", title: "Redux, RTK" },
    { id: 8, locale: "en", documentId: "8", title: "React Query" },
    { id: 9, locale: "en", documentId: "9", title: "React Hook Form" },
    { id: 10, locale: "en", documentId: "10", title: "Zod" },
  ],
  id: 1,
  locale: "en",
  documentId: "1",
  title: "Projects",
  componentId: "section-skills",
  __component: "dynamic-zone.section-skills",
  button: {
    id: 1,
    documentId: "1",
    title: "See more",
    subtitle: "Projects",
    isExternal: false,
    href: "#",
  },
};

export const mockStrapiText = {
  id: 1,
  locale: "en",
  documentId: "1",
  title: "Projects",
  componentId: "section-text",
  __component: "dynamic-zone.section-text",
  content:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean luctus dictum iaculis. Proin suscipit lorem porttitor odio gravida rhoncus. Praesent a sollicitudin massa. Curabitur convallis volutpat efficitur. Donec auctor rhoncus magna non consequat. Aenean sodales lacinia lorem, sit amet porttitor nibh laoreet ut. Vivamus eleifend sit amet turpis eu cursus. Nulla consequat purus lorem, quis faucibus ipsum lobortis nec. Vivamus vel mauris eget libero consectetur tincidunt. Nulla nibh libero, condimentum quis eros in, dictum suscipit quam.",
  button: {
    id: 1,
    documentId: "1",
    title: "See more",
    subtitle: "Projects",
    isExternal: false,
    href: "#",
  },
};

export const mockStrapiProjects = {
  id: 1,
  locale: "en",
  documentId: "1",
  title: "Projects",
  componentId: "section-projects",
  __component: "dynamic-zone.section-projects",
  button: {
    id: 1,
    documentId: "1",
    title: "See more",
    subtitle: "Projects",
    isExternal: false,
    href: "#",
  },
  projects: [
    {
      id: 1,
      documentId: "1",
      title: "TODO App",
      stack: "NextJS, TailwindCSS, TypeScript",
      githubURL: null,
      projectURL: "https://shteyn-web.ru",
      preview: mockStrapiImage,
    },
    {
      id: 2,
      documentId: "2",
      title: "Landing",
      stack: "NextJS, TailwindCSS, TypeScript",
      githubURL: "https://github.com/xN8Tx/portfolio",
      projectURL: null,
      preview: mockStrapiImage,
    },
    {
      id: 3,
      documentId: "3",
      title: "E-Commerce",
      stack: "NextJS, TailwindCSS, TypeScript",
      githubURL: "https://github.com/xN8Tx/portfolio",
      projectURL: "https://shteyn-web.ru",
      preview: mockStrapiImage,
    },
    {
      id: 4,
      documentId: "4",
      title: "E-Commerce",
      stack: "NextJS, TailwindCSS, TypeScript",
      githubURL: null,
      projectURL: null,
      preview: mockStrapiImage,
    },
  ],
};

export const strapiMockPageData = {
  data: [
    {
      dynamicZone: [
        mockStrapiHero,
        mockStrapiText,
        mockStrapiSkills,
        mockStrapiProjects,
      ],
    },
  ],
};
