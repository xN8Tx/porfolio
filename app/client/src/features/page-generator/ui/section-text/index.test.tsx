import { screen } from "@testing-library/react";
import { render } from "@/shared/test-lib";

import { mockStrapiHero, mockStrapiText } from "../../mocks";

import { SectionText } from ".";

describe("Page Generator/Section Text", () => {
  test("shold render", () => {
    render(<SectionText {...mockStrapiText} />);

    const title = screen.getByTestId("page-text_title");
    expect(title).toBeInTheDocument();
    expect(title).toHaveTextContent(mockStrapiText.title);

    const subtitle = screen.getByTestId("page-text_content");
    expect(subtitle).toBeInTheDocument();
    expect(subtitle).toHaveTextContent(mockStrapiText.content);

    const button = screen.getByTestId("page-text_button");
    expect(button).toBeInTheDocument();
  });
  test("should match snapshot", () => {
    const tree = render(<SectionText {...mockStrapiText} />);

    expect(tree).toMatchSnapshot();
  });
});
