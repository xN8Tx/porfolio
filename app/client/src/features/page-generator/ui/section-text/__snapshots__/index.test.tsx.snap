// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Generator/Section Text should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <script
        nonce=""
      >
        ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("data-theme","theme","dark",null,["light","dark"],null,true,true)
      </script>
      <section
        class="about"
        data-scroll-section="true"
        data-testid="page-text"
        id="section-text"
      >
        <h3
          class="ui-heading small light"
          data-testid="page-text_title"
        >
          Projects
        </h3>
        <p
          class="ui-paragraph large default"
          data-scroll="true"
          data-testid="page-text_content"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean luctus dictum iaculis. Proin suscipit lorem porttitor odio gravida rhoncus. Praesent a sollicitudin massa. Curabitur convallis volutpat efficitur. Donec auctor rhoncus magna non consequat. Aenean sodales lacinia lorem, sit amet porttitor nibh laoreet ut. Vivamus eleifend sit amet turpis eu cursus. Nulla consequat purus lorem, quis faucibus ipsum lobortis nec. Vivamus vel mauris eget libero consectetur tincidunt. Nulla nibh libero, condimentum quis eros in, dictum suscipit quam.
        </p>
        <button
          class="ui-link__button hover regular default"
          data-testid="ui-button_button"
        >
          <a
            data-testid="page-text_button"
            href="#"
            target="_self"
          >
            <p
              data-testid="ui-button_paragraph"
            >
              See more
            </p>
            <p
              data-testid="ui-button_subtitle"
            >
              Projects
            </p>
          </a>
        </button>
      </section>
    </div>
  </body>,
  "container": <div>
    <script
      nonce=""
    >
      ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("data-theme","theme","dark",null,["light","dark"],null,true,true)
    </script>
    <section
      class="about"
      data-scroll-section="true"
      data-testid="page-text"
      id="section-text"
    >
      <h3
        class="ui-heading small light"
        data-testid="page-text_title"
      >
        Projects
      </h3>
      <p
        class="ui-paragraph large default"
        data-scroll="true"
        data-testid="page-text_content"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean luctus dictum iaculis. Proin suscipit lorem porttitor odio gravida rhoncus. Praesent a sollicitudin massa. Curabitur convallis volutpat efficitur. Donec auctor rhoncus magna non consequat. Aenean sodales lacinia lorem, sit amet porttitor nibh laoreet ut. Vivamus eleifend sit amet turpis eu cursus. Nulla consequat purus lorem, quis faucibus ipsum lobortis nec. Vivamus vel mauris eget libero consectetur tincidunt. Nulla nibh libero, condimentum quis eros in, dictum suscipit quam.
      </p>
      <button
        class="ui-link__button hover regular default"
        data-testid="ui-button_button"
      >
        <a
          data-testid="page-text_button"
          href="#"
          target="_self"
        >
          <p
            data-testid="ui-button_paragraph"
          >
            See more
          </p>
          <p
            data-testid="ui-button_subtitle"
          >
            Projects
          </p>
        </a>
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
